frontend/src/components/CalcSpectrum.tsx

```diff
@@ -41,6 +42,7 @@ export const CalcSpectrum: React.FC = () => {
   const [useGesia, setUseGesia] = useState(false);
   const [useSlit, setUseSlit] = useState(false); // checking that user wants to apply the slit function or not in available modes
   const [useSimulateSlitFunction, setUseSimulateSlitFunction] = useState(false); // checking the mode and enable or disable slit feature
+  const [downloadbutton, setdownloadbutton] = useState(false);
```

Comment:
Should be camelCase for hook variables, so `downloadButton` and `setDownloadButton`